benchmark.js

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var process = require('process');
+var setTimeout = require('timers').setTimeout;
+var BenchmarkLogger = require('./benchmark-logger');
+
+function Benchmark(options) {
+    if (!(this instanceof Benchmark)) {
+        return new Benchmark(options);
+    }
+
+    var self = this;
+
+    self.tcurl = options.tcurl;
+    self.logger = options.logger;
+    self.delegate = new BenchmarkLogger();
```

Comment:
It does not make sense to have both a logger and a benchmark logger. I suggest consolidating these into one delegate and instantiate the benchmark delegate instead of the logging delegate.